lib/local_daily_digest.js

```diff
@@ -30,18 +39,65 @@ function get(masterCb) {
 				lastDigestSent = new Date(lastDigestSent);
 				var offset = current.valueOf() - lastDigestSent.valueOf();
 				if (offset > 24 * 60 * 60 * 1000) { // If last digest was sent more than 24 hours ago
-	  				allLocalProfilesWDigest.push(profile);
-					// For all the profiles which want to get a digest, get a list of the locations they want a digest for
-					profile.dailyDigest.forEach(function (location) {
-						requiredLocationsId.push(location);
+					//if profile roles does not contain manager or editor
+
+	  				//forEach country
+	  				profile.dailyDigest.forEach(function (location) {
+		  				profile.roles.forEach(function(role){
+	  						if(role.split(/:(.+)?/)[1] != 'undefined'){
+			  					if(location == role.split(/:(.+)?/)[1] && (role.split(/:(.+)?/)[0] == 'manager' || role.split(/:(.+)?/)[0] == 'editor')){
+					  				allLocalProfilesWDigest.push(profile);
+									requiredLocationsId.push(location);
+			  					}
+			  					else {
+			  						allLocalProfilesWOrgDigest.push(profile);
+									reqLocIdOrgDigest.push(location);
+			  					}
+		  					}		
+	  					})
 					});
 				}
 		  	});
+		  	return cb();
+			});
+		},
+		function (cb){
+			// console.log(allLocalProfilesWDigest ,allLocalProfilesWOrgDigest);
+			async.forEachSeries(reqLocIdOrgDigest, function(location, callback2) {
+				// console.log(location);
+				Contact.find({$query: {type: 'local', locationId: location, created: {$gt: Date.now() - (24*60*60*1000)}, status: true, $and: [{organization: {$not: {$size: 0}}}, {organization: {$ne : null}}]}, $orderby : {location: -1} }, function (err, contacts){
+					async.forEachSeries(contacts, function(contact, callback3){
+						if(!contact){
+							console.log("No users found1");
+							return cb();
+						}
 
-			console.log("the countries these managers/editors want the dailyDigest for are: ", requiredLocationsId);
+						Contact.find({$query: {type: 'local', locationId: location, created: {$gt: Date.now() - (24*60*60*1000)},  status: true , 'organization.name': contact.organization[0].name}, $orderby : {location: -1}} , function (err, orgContacts){
```

Comment:
I think there is an issue with the logic here. First you query for all contacts where organization != null, then for each of these contacts you query for contacts with the same organization. So first, you're making the same query multiple times. Then, below when you send the email, you don't check anywhere which organization the current profile belongs to to send it the right org digest...

The algorithm should be as follows: 
foreach profile with a dailyDigest {
  foreach dailyDIgest location {
    if profile is a manager or editor of this location {
      send full daily digest;
    }
    else {
      Get the profile's organization from his global profile;
      if (!cacheOrganizations[profileOrganization].length) {
        Query the contacts with organization == profileOrganization;
         cacheOrganizations[profileOrganization] = organizations;
       }
    }
  }
}